File: src/main/kotlin/strings/GoalParserInterpretation.java
package strings;

public class GoalParserInterpretation {

    public static void main(String[] args) {
      String command = "G()(al)";
      interpret(command);
    }

    static  public String interpret(String command) {
        return command.replace("()","o").replace("(al)","al");
    }
}

File: strings/GoalParserInterpretation.kt
package strings

import array.removeElement

// command consists of "G", "()", and/or "(al)" in some order.
/**
G -> G
() -> o
(al) -> al
The final concatenated result is "Goal".
 */
fun main() {
    var command: String = "G()(al)"
    println(interpret(command = command))

}

fun interpret(command: String) = command.replace("()", "o").replace("(al)", "al")

